File: payroll-api/src/main/java/com/company/payroll/mapper/StaffPromotionMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffPromotion;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffPromotionMapper {
    int deleteByPrimaryKey(Integer pId);
    int insert(StaffPromotion row);
    int insertSelective(StaffPromotion row);
    StaffPromotion selectByPrimaryKey(Integer pId);
    List<StaffPromotion> selectList();
    List<StaffPromotion> selectListByStaffId(Integer staffId);
    int updateByPrimaryKey(StaffPromotion row);
    int updateByPrimaryKeySelective(StaffPromotion row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffPromotionMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffPromotion
import org.springframework.stereotype.Repository

@Repository
interface StaffPromotionMapper {
  fun deleteByPrimaryKey(pId: Int): Int

  fun insert(row: StaffPromotion): Int

  fun insertSelective(row: StaffPromotion): Int

  fun selectByPrimaryKey(pId: Int): StaffPromotion?

  fun updateByPrimaryKeySelective(row: StaffPromotion): Int

  fun updateByPrimaryKey(row: StaffPromotion): Int
}